ui/src/app/applications/components/application-status-panel/application-status-panel.tsx

```diff
@@ -59,6 +59,8 @@ const sectionHeader = (info: SectionInfo, onClick?: () => any) => {
 };
 
 const hasRollingSyncEnabled = (application: models.Application): boolean => {
+    // Only show Progressive Sync for applications that have an ApplicationSet owner reference
+    // The actual strategy validation is done in the ProgressiveSyncStatus component
     return application.metadata.ownerReferences?.some(ref => ref.kind === 'ApplicationSet') || false;
 };
```

Comment:
Hi @aali309

It’s me from the reported issue: https://github.com/argoproj/argo-cd/issues/24190#issuecomment-3200543083 :)

Maybe I’m missing something, but wouldn’t it be more fitting to rename this function to `hasApplicationSetParent` or `isPartOfApplicationSet` or something similar?

The function only checks if there’s an owner reference pointing to an ApplicationSet, and does not check if _rolling sync has been enabled_. I understand that the sync-strategy is checked later in the code, but I would argue that the current function name is a bit misleading considering what it actually does.